db/migrate/20190729214242_add_ipaddresses_subnets_and_networks.rb

```diff
@@ -0,0 +1,150 @@
+class AddIpaddressesSubnetsAndNetworks < ActiveRecord::Migration[5.2]
+  def change
+    create_table :networks, id: :bigserial, force: :cascade do |t|
+      t.references :tenant, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source, :index => false, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source_region, :index => true, :null => true, :foreign_key => {:on_delete => :cascade}
+      t.references :subscription, :index => true, :null => true, :foreign_key => {:on_delete => :cascade}
+      t.references :orchestration_stack, :index => true, :null => true, :foreign_key => {:on_delete => :cascade}
+
+      t.string :source_ref, :null => false
+      t.string :name
+      t.string :cidr
+      t.string :status
+
+      t.jsonb :extra
+
+      t.datetime :resource_timestamp
+      t.jsonb :resource_timestamps, default: {}
+      t.datetime :resource_timestamps_max
+
+      t.timestamps
+      t.datetime :archived_at
+      t.datetime :source_created_at
+      t.datetime :source_deleted_at
+      t.datetime :last_seen_at
+
+      t.index :archived_at
+      t.index :last_seen_at
+
+      t.index %i[source_id source_ref], :unique => true
+    end
+
+    create_table :subnets, id: :bigserial, force: :cascade do |t|
+      t.references :tenant, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source, :index => false, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source_region, :index => true, :null => true, :foreign_key => {:on_delete => :cascade}
+      t.references :subscription, :index => true, :null => true, :foreign_key => {:on_delete => :cascade}
+      t.references :orchestration_stack, :index => true, :null => true, :foreign_key => {:on_delete => :cascade}
+
+      t.references :network, :index => true, :null => true, :foreign_key => {:on_delete => :cascade}
+
+      t.string :source_ref, :null => false
+      t.string :name
+      t.string :cidr
+      t.string :status
+
+      t.jsonb :extra
+
+      t.datetime :resource_timestamp
+      t.jsonb :resource_timestamps, default: {}
+      t.datetime :resource_timestamps_max
+
+      t.timestamps
+      t.datetime :archived_at
+      t.datetime :source_created_at
+      t.datetime :source_deleted_at
+      t.datetime :last_seen_at
+
+      t.index :archived_at
+      t.index :last_seen_at
+
+      t.index %i[source_id source_ref], :unique => true
+    end
+
+    create_table :ipaddresses, id: :bigserial, force: :cascade do |t|
```

Comment:
@agrare shouldn't it be `ip_addresses` ? OR do you like `ipaddresses` more?

Reply:
I think `ipaddresses` is fine